### PR TITLE
[DUOS-1657][risk=no] Add library card validation to DAR submission

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1657

Adds a validation check on the researcher having a library card.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/dar_application/DataUseAgreements.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const StepAlertTemplate = (props) => {
 
   return (
     ul({style: ulLinkStyle}, [
-      li({isRendered: props.step1Invalid}, [
+      li({isRendered: props.step1Invalid || props.libraryCardInvalid}, [
```
Comment (shaemarks) [inline src/pages/dar_application/DataUseAgreements.js:18]: Works properly with missing library card, so this comment is potentially out of scope for this ticket: 

When my profile is incomplete (I removed part of my address), I get the alert on the Step 1 page itself to complete my profile, but I do not get this link at the end of the dar application to fix Step 1. I think `props.step1Invalid` doesn't check for all ways a profile may be incomplete. Not sure on the intended behavior, but I thought I would mention it.

**Review by shaemarks — COMMENTED** [review]
I left a comment on a potentially missing alert at the end of the dar application when I have a library card but an incomplete profile. Otherwise, looks good to me for researchers with and without library cards.

**Review by quazi-h — APPROVED** [review]
Changes look good to me and the UI is functioning as expected with these changes. I loaded this branch to a local instance and went through the process of trying to create a DAR both with and without a library card. I was unable to create a DAR without a library card and observed the correct warning text and links when I attempted to do so. I did observe the same thing Shae did - in my case I tried to remove the eraCommonsId. Is that piece supposed to be a required part of the profile?

File: src/pages/dar_application/DataUseAgreements.js (hunk 1 above)
Comment (rushtong) [reply src/pages/dar_application/DataUseAgreements.js:18]: I can reproduce this behavior in dev, so I'm going to file a separate bug ticket for: 
> When my profile is incomplete (I removed part of my address), I get the alert on the Step 1 page itself to complete my profile, but I do not get this link at the end of the dar application to fix Step 1. I think props.step1Invalid doesn't check for all ways a profile may be incomplete.

Comment (rushtong) [reply src/pages/dar_application/DataUseAgreements.js:18]: See https://broadworkbench.atlassian.net/browse/DUOS-1685

**Pull request merged**